src/quemb/kbe/fragment.py

```diff
@@ -167,6 +171,40 @@ def fragmentate(
             iao_valence_basis=iao_valence_basis,
             kpt=kpt,
         )
-
+    elif frag_type == "chemgen":
+        warn("Periodic BE1 with chemgen is a temporary solution.")
+        if kpt is None:
```

Comment:
I would put all the checks for invariants as early as possible.
This would also fix the surprising output when having `be_type != "be1"`.
Currently it would first warn
"Periodic BE1 with chemgen is a temporary solution."
and then 
raise the error that only `be_type = 'be1'` is supported.